Unable to enter a new PCR product

After the merge of #20 , Hong has been unable to submit new PCR products. The relevant errors in the log:

```
2019-04-24 15:25:50 [26171] [ERROR] Viroverse: HTTP 500: Error: Couldn't save PCR product
2019-04-24 15:27:35 [26173] [ERROR] Viroverse.Controller.input: Couldn't save PCR product: obj(DBIx::Class::Storage::DBI::_dbh_execute(): DBI Exception: DBD::Pg::st execute failed: ERROR:  date/time field value out of range: "2019-27-02"
HINT:  Perhaps you need a different "datestyle" setting. [for Statement "INSERT INTO viroserve.pcr_template ( date_completed, sample_id, scientist_id, unit_id, volume) VALUES ( ?, ?, ?, ?, ? ) RETURNING pcr_template_id" with ParamValues: 1='2019-27-02', 2='106839', 3='245', 4='11', 5='6'] at /opt/home/vverse/viroverse/lib/Viroverse/Controller/input.pm line 470
)
2019-04-24 15:27:35 [26173] [ERROR] Viroverse: HTTP 500: Error: Couldn't save PCR product
2019-04-24 15:30:15 [26179] [ERROR] Viroverse.Controller.input: Couldn't save PCR product: obj(DBIx::Class::Storage::DBI::_dbh_execute(): DBI Exception: DBD::Pg::st execute failed: ERROR:  invalid input syntax for integer: "" [for Statement "INSERT INTO viroserve.pcr_product ( date_completed, endpoint_dilution, enzyme_id, notes, pcr_template_id, protocol_id, replicate, round, scientist_id) VALUES ( ?, ?, ?, ?, ?, ?, ?, ?, ? ) RETURNING pcr_product_id" with ParamValues: 1='2019-02-27', 2=undef, 3='20', 4='', 5='406737', 6='', 7='1', 8='1', 9='25'] at /opt/home/vverse/viroverse/lib/Viroverse/Controller/input.pm line 478
)
2019-04-24 15:30:15 [26179] [ERROR] Viroverse: HTTP 500: Error: Couldn't save PCR product
```

I verified that she was entering the data correctly; the date/time one is especially strange as she was definitely entering the date as 2019-02-27 although that seems to be separate from the actual insertion error.

Per the log it appears that `protocol_id` is somehow blank.

--- a/lib/Viroverse/Controller/input.pm
+++ b/lib/Viroverse/Controller/input.pm
@@ -101,7 +101,7 @@ sub extraction_post : Local {
 
     my %resolve = (
         unit_id => 'unit_id',
-        scientist_id => 'scientist_id', 
+        scientist_id => 'scientist_id',
         extract_type_id => 'extract_type_id',
         concentration_unit_id => 'unit_id',
         eluted_vol_unit_id => 'unit_id'
@@ -197,9 +197,9 @@ sub rt_add : Local {
     foreach my $field_name (@required) {
             $context->detach('user_error',["$field_name is a required field","missing $field_name"]) unless length($params{$field_name}) > 0  ;
     }
-    if ($params{ratio_toggle} eq "special" 
-            and not (length($params{ratio_special_rna}) 
-                and length($params{ratio_special_cdna}))) 
+    if ($params{ratio_toggle} eq "special"
+            and not (length($params{ratio_special_rna})
+                and length($params{ratio_special_cdna})))
     {
         $context->detach('user_error',["Special RNA:cDNA ratio must be specified", "missing ratio element".length($params{ratio_special_rna})]);
     }
@@ -240,14 +240,14 @@ sub rt_add : Local {
             } else {
                 $used_primers{$primer_name} = 1;
             }
-            my @primer_results = Viroverse::Model::primer->search({ name => $primer_name } ) ; 
+            my @primer_results = Viroverse::Model::primer->search({ name => $primer_name } ) ;
             if (@primer_results ) {
                 $rt->add_to_primers({primer_id => $primer_results[0]});
             } else {
                 $context->detach('user_error',["Could not find primer ".$primer_name,'']);
             }
 
-        } 
+        }
 
         push @new_rt, $rt->give_id;
     }
@@ -274,7 +274,7 @@ sub bisulfite_conversion : Local {
 
 sub bisulfite_conversion_add : Local {
     my ($self, $context) = @_;
-    
+
     my %params = %{$context->req->params};
     my @template_params = grep /box$/, keys %params;
     my ($scientist) = Viroverse::Model::scientist->search({ name => $params{scientist_name} });
@@ -479,7 +479,7 @@ sub pcr_add : Local {
                         pcr_template      => $template,
                         scientist         => $pcr_scientist,
                         date_completed    => $round->{completed_date},
-                        protocol_id       => $round->{protocol_id},
+                        protocol_id       => $round->{protocol_id} || undef,
                         enzyme_id         => $round->{enzyme_id},
                         endpoint_dilution => $round->{endpoint},
                         notes             => $round->{notes},
@@ -563,7 +563,7 @@ sub receive_gel : Local {
     }
 
     my $gel = Viroverse::Model::gel->insert({
-        image=>$img, 
+        image=>$img,
         mime_type => $mime,
         name => $upload->filename,
         scientist_id => $context->stash->{scientist}->scientist_id
@@ -644,7 +644,7 @@ sub pos_pcr_add : Local {
 
         my $p_conc = $params{purif_final_conc};
         if (length($p_conc) > 0 ) {
-            $context->detach('user_error',["Illegal purification final concentration"]) 
+            $context->detach('user_error',["Illegal purification final concentration"])
                 if $p_conc !~ m/^(\d+(\.\d+)?)$|^(\.\d+)$/;
             $new_purif{final_conc} = $p_conc;
         }

